modules/@apostrophecms/piece-type/index.js

```diff
@@ -388,20 +395,61 @@ module.exports = {
       },
       'apostrophe:modulesRegistered': {
         composeBatchOperations() {
-          self.batchOperations = Object.keys(self.batchOperations).map(key => ({
-            action: key,
-            ...self.batchOperations[key]
-          })).filter(batchOperation => {
-            // If a `requiredField` is registered, only include the operation
-            // if that field is present on the schema.
-            if (batchOperation.requiredField && !_.find(self.schema, {
-              name: batchOperation.requiredField
-            })) {
-              return false;
+          const groupedOperations = Object.entries(self.batchOperations)
+            .reduce((acc, [ opeName, properties ]) => {
+              const requiredFieldNotFound = properties.requiredField && !self.schema
+                .some((field) => field.name === properties.requiredField);
+
+              if (requiredFieldNotFound) {
+                return acc;
+              }
+
+              const associatedGroup = getAssociatedGroup(opeName);
+              const currentOperation = {
+                name: opeName,
+                ...properties
+              };
+              const { name, ...props } = getCurrentOrGroup(
+                currentOperation,
+                associatedGroup,
+                acc
+              );
+
+              return {
+                ...acc,
+                [name]: {
+                  ...props
+                }
+              };
+            }, {});
+
+          self.batchOperations = Object.entries(groupedOperations)
+            .map(([ name, properties ]) => ({
+              name,
+              ...properties
+            }));
+
+          function getCurrentOrGroup (currentOpe, [ groupName, groupProperties ], acc) {
```

Comment:
Could we call this `getOperationOrGroup`? "Current" makes sense in the context of the reducer, but doesn't communicate what this is doing as well.

Reply:
Yes right, I modify it.